public/assets-sync.js

```diff
@@ -245,35 +265,40 @@ class ForgeAssetSync {
                 // If there is, jump to the reconcile method
                 if (localFileExists) {
                     result = await this.reconcileLocalMatch(asset);
+                    if (!result) {
+                        throw new Error(`Forge VTT | Asset Sync: could not reconcile local match for ${asset.name}`);
+                    }
                 } else if (localDirMissing) {
                     // If the local directory couldn't be created, treat it as a failed sync
-                    result = false;
+                    throw new Error(`Forge VTT | Asset Sync: local directory missing for ${asset.name}`);
                 } else {
                     // If not, the asset needs to be fully synced
                     result = await this.syncAsset(asset);
+
+                    if (!result) {
+                        throw new Error(`Forge VTT | Asset Sync: could not sync asset for ${asset.name}`);
+                    }
```

Comment:
We should either run `updateProgress` before throwing, or run it in the catch block below.

Reply:
See update, I added a `finally` block to take care of the last parts of this. Though I wonder why we are updating the progress *after* syncing the asset? Should that maybe go above?